docs/guide/in-depth/advanced-chunks.md

```diff
@@ -65,3 +297,57 @@ When we run `node ./main.js`, the traversal order of the modules would be `main.
 `second.js` tries to call `__esm` function before it gets initialized. This will lead to a runtime error which is trying to call `undefined` as a function.
 
 With forcefully generated `runtime.js`, the bundler ensures any chunk that depends on runtime code would first load `runtime.js` before executing itself. This guarantees that the runtime code is always executed before any other chunks, preventing circular import issues.
+
+### Why the dependencies of the captured module get captured too?
+
+When a module is captured by a group, rolldown will try to capture its dependencies recursively. This is because rolldown is only allowed to mangling exports of non-entry chunks by default.
+
+For example, if you have the following code:
+
+```js
+// entry.js
+import { value } from './a.js';
+
+console.log(value);
+
+export const foo = 'foo';
+
+// a.js
+import { value as valueB } from './b.js';
+export const value = 'a' + valueB;
+
+// b.js
+export const value = 'b';
+```
+
+Let's say we want to move the `a.js` module into a separate chunk with keeping the `b.js` in the same chunk as `entry.js`. We get
+
+:::code-group
+
+```js [entry.js]
+import { value } from './a.js';
+
+// b.js
+const value = 'b';
+
+// entry.js
+const foo = 'foo';
+console.log(value);
+
+export { foo, value };
+```
+
+```js [a.js]
+import { value } from './entry.js';
+
+// a.js
+export const value = 'a' + value;
+```
+
+:::
+
+You could see, to make `a.js` work, we have to change the export signature of the entry chunk `entry.js` and add an additional export `value`. This totally violates the original intention of the code, which is to only export `foo` from `entry.js`.
+
+Fortunately, rolldown supports [`InputOptions.preserveEntrySignatures: 'allow-extension'`](/reference/config-options#preserveentrysignatures) to let you inform the bundler that it is allowed to change the export signature of the entry chunk.
+
+Enabling [`InputOptions.preserveEntrySignatures: 'allow-extension'`](/reference/config-options#preserveentrysignatures) will prevent the bundler from capturing the dependencies of the captured module.
```

Comment:
Just to confirm, advanced chunks cannot solve "hash cascading problem", right?

Reply:
Yeah. If I understand you correctly, it should be solved by future `importmap` support.